torch/signal/windows/__init__.py

```diff
@@ -0,0 +1,159 @@
+import warnings
+
+from .windows import cosine, exponential, gaussian
+
+from torch._torch_docs import factory_common_args
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+
+def _add_docstr(function, docstr):
+    function.__doc__ = docstr
+
+
+_add_docstr(
+    cosine,
+    r"""
+Computes a window with a simple cosine waveform.
+Also known as the sine window.
+
+The cosine window is defined as follows:
+
+.. math::
+    w(n) = \cos{\left(\frac{\pi n}{M} - \frac{\pi}{2}\right)} = \sin{\left(\frac{\pi n}{M}\right)}
+
+Where `M` is the length of the window.
+    """ +
+    r"""
+
+Args:
+    window_length (int): the length of the output window.
+        In other words, the number of points of the cosine window.
+    periodic (bool, optional): If `True`, returns a periodic window suitable for use in spectral analysis.
+        If `False`, returns a symmetric window suitable for use in filter design. Default: `True`.
+
+Keyword args:
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples:
+    >>> # Generate a cosine window without keyword args.
+    >>> torch.signal.windows.cosine(10)
+    tensor([0.1423, 0.4154, 0.6549, 0.8413, 0.9595, 1.0000, 0.9595, 0.8413, 0.6549,
+    0.4154])
+
+    >>> # Generate a symmetric cosine window.
+    >>> torch.signal.windows.cosine(10,periodic=False)
+    tensor([0.1564, 0.4540, 0.7071, 0.8910, 0.9877, 0.9877, 0.8910, 0.7071, 0.4540,
+    0.1564])
+
+.. note::
+    The window is normalized to 1 (maximum value is 1), however, the 1 doesn't appear if `M` is even
+    and `periodic` is `False`.
```

Comment:
Thank you for how thorough these docs are! A small nit: We usually put `note`s before `Args` so that they are more visible. For these style conventions, have a browse through the `linalg` docs :)

Reply:
Yes, I saw other examples but it was a bit awkward to my eyes, but sure, let's do it that way!